Name: Shreyas Dongre   
Roll no: 082        
Division: I                         
Semester: 2nd Semester                                           
Batch: I3                                              
Experiment 1- To study components of a python program.                                     
Date of Experiment: 11/2/2021 

## Task 1:  
Program to demonstrate data type in python.                           
1.1 Program to print basic data type of given variable                        
i. a = 5	                                                                 	 ii. a = 2.0 	
iii. a = 1+2j 	
iv. a = ‘NMIMS’


In [ ]:
a=5
print(type(a))
a=2.0
print(type(a))
a=1+2j
print(type(a))
a='NMIMS'
print(type(a))


<class 'int'>
<class 'float'>
<class 'complex'>
<class 'str'>


##Task 2:  
Write a Python Program to the roots of a Quadratic Equation. d = b2 - 4ac


In [ ]:
import cmath
a=int(input('Enter a '))
b=int(input('Enter b '))
c=int(input('Enter c '))

d=(b**2)-4*(a)*(c)
root1=(-b+cmath.sqrt(d))/(2*a)
root2=(-b-cmath.sqrt(d))/(2*a)

print('Roots of Quadratic Equation are {} and {}'.format(root1,root2))

Enter a 10
Enter b 20
Enter c 25
Roots of Quadratic Equation are (-1+1.224744871391589j) and (-1-1.224744871391589j)


##Task 3:  
Write a Python Program to Convert Celsius To Fahrenheit.


In [ ]:
c=float(input('Enter degree in celcius '))
f=(c*(9/5))+ 32
print('Fahrenheit temperature is:',f)

Enter degree in celcius 9
Fahrenheit temperature is: 48.2


##Task 4: 
What will be the output of the following Python code? Justify.

1.	str="NMIMS BTech Artificial Intelligence"
str[11:22]


2.	a=10
a = a + '2'
a = a*2


3.	x = 1234
print("%08d"%x)


In [ ]:
#1
str="NMIMS Btech Artificial Intelligence"
str[11:22]

' Artificial'

**Ans:** The output was ' Artificial' because of str[11:22] which ia basically slicing which means that the string got sliced from 11th index position (which is the space before the word artificial) to 21st index position(that is till the letter 'l').

In [ ]:
#2
a=10
print(type(a))
a=a+'2'
print(type(a))
a=a*2
print(type(a))

<class 'int'>


TypeError: ignored

**Ans:** The output for the first one was given as 'int'. But for the second one there was an error because python can not  add int and string.

In [ ]:
#3
x = 1234 
print("%08d"%x)

00001234


**Ans:** The output is 00001234 because of %08d . It is a length specifier. As %08d is writeen, the length increases  to 8 places and hence the output 00001234. 

##Task 5:  
Write a Python program to convert the distance (in feet) to inches, yards, and miles.


In [ ]:
f=float(input('Enter distance in feet '))
i=f*12
print('Inches',i)
y=f/3
print('Yards',y)
m=f/5280
print('Miles',m)

Enter distance in feet 10
Inches 120.0
Yards 3.3333333333333335
Miles 0.001893939393939394


##Task 6 : 
Write a Python program to solve (x + y) * (x + y). Read the values of x and y from the user.
Expected Output : ((x + y) ^ 2)


In [ ]:
x=float(input('Enter x '))
y=float(input('Enter y '))

ans=(x+y)*(x+y)
print("(({}+{})^2)={}".format(x,y,ans))

Enter x 3
Enter y 4
((3.0+4.0)^2)=49.0


##Task 7 :
1.	Write a Python program to convert seconds to day, hour, minutes and seconds.
2.	Write a Python program to convert the number of days to years, months & days.


In [ ]:
#1
seconds=float(input('Enter number of seconds'))
day=int(seconds/86400)
a=(seconds%86400)
hours=int(a/3600)
b=(a%3600)
minutes=int(b/60)
seconds=(b%60)
print('Days:', day)
print('Hours:',hours)
print('Minutes',minutes)
print('seconds',seconds)


Enter number of seconds989998
Days: 11
Hours: 10
Minutes 59
seconds 58.0


In [ ]:
#2
days=float(input('Enter no of days  '))
years= int(days/365)
a=(days%365)
months=int(a/30)
b=(a%30)
days=int(b)

print('No of years',years)
print('No of months',months)
print('No of days',days)

Enter no of days 1099
No of years 3
No of months 0
No of days 4


##Task 8 : 

1.	Write a Python Program to Calculate Area of an Ellipse.
2.	Write a Python Program to Find the Surface Area of Cone.



In [ ]:
#1
import math
a=int(input('Enter len of major axis '))
b=int(input('Enter len of minor axis '))
area=math.pi*a*b
print("Area of ellipse ",area)

Enter len of major axis 15
Enter len of minor axis 10
Area of ellipse  471.23889803846896


In [ ]:
#2
import math
r=int(input('Enter radius of cone '))
l=int(input('Enter len of cone '))
area=math.pi*r*(r+l)
print("Total Surface Area of cone: ",area)

Enter radius of cone 12
Enter len of cone 17
Total Surface Area of cone:  1093.274243449248


##Task 9 : 
Write a Python program to display your birth date.            
Eg : birth_date = (11, 12, 2014)           
Sample Output: Your Birthdate is  : 11 / 12 / 2014



In [ ]:
d=(11,12,2014)
print('Your Birthdate is: %i / %i / %i'%d)

Your Birthdate is: 11 / 12 / 2014


##CONCLUSION:
1)Used basic functions like print(),input()            
2)Find the datatype of variables.                   
3)Using various operators.                       
4)using various operators like arithmetic & assignment operators on different   data types.                      
5)Importing libraries like math and cmath for doing complex mathematical calculations and complex number calculations.            
6)Using Format Function.         
7)Using  Tuples.           
8)Using length specifier for a datatype.